Description: Fix build of WebRTC on non-Linux systems
 * Define the WEBRTC_LINUX macro only on Linux, and not on GNU/Hurd or FreeBSD.
 * Fix type cast in the CurrentThreadId function.
Author: Nicholas Guriev <guriev-ns@ya.ru>
Last-Update: Sat, 25 Jul 2020 10:57:07 +0300

--- a/Makefile.am
+++ b/Makefile.am
@@ -383,7 +383,7 @@ SRC += \
 webrtc_dsp/rtc_base/logging_mac.mm \
 webrtc_dsp/rtc_base/logging_mac.h
 else
-CFLAGS += -DWEBRTC_LINUX
+CFLAGS += $(if $(filter linux%,@host_os@),-DWEBRTC_LINUX)
 endif
 
 if TARGET_CPU_X86
--- a/libtgvoip.gyp
+++ b/libtgvoip.gyp
@@ -869,20 +869,18 @@
             '"<(OS)" == "linux"', {
               'defines': [
                 'WEBRTC_POSIX',
-                'WEBRTC_LINUX',
               ],
               'conditions': [
                 [ '"<!(uname -m)" == "i686"', {
                   'cflags_cc': [
                     '-msse2',
                   ],
+                }], ['"<!(uname -s)" == "Linux"', {
+                  'defines': [
+                    'WEBRTC_LINUX',
+                  ],
                 }]
               ],
-              'direct_dependent_settings': {
-                'libraries': [
-
-                ],
-              },
             },
           ],
         ],
--- a/webrtc_dsp/rtc_base/platform_thread_types.cc
+++ b/webrtc_dsp/rtc_base/platform_thread_types.cc
@@ -31,7 +31,7 @@ PlatformThreadId CurrentThreadId() {
   return syscall(__NR_gettid);
 #else
   // Default implementation for nacl and solaris.
-  return reinterpret_cast<pid_t>(pthread_self());
+  return static_cast<pid_t>(pthread_self());
 #endif
 #endif  // defined(WEBRTC_POSIX)
 }
